libpod/runtime_volume.go

```diff
@@ -141,3 +141,55 @@ func (r *Runtime) PruneVolumes(ctx context.Context, filterFuncs []VolumeFilter)
 	}
 	return preports, nil
 }
+
+// PruneVolumesWithOptions removes unused volumes from the system with options
+func (r *Runtime) PruneVolumesWithOptions(ctx context.Context, filterFuncs []VolumeFilter, includePinned bool) ([]*reports.PruneReport, error) {
+	preports := make([]*reports.PruneReport, 0)
+	vols, err := r.Volumes(filterFuncs...)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, vol := range vols {
+		// Skip pinned volumes unless explicitly requested
+		if vol.Pinned() && !includePinned {
+			continue
+		}
+
+		report := new(reports.PruneReport)
+		volSize, err := vol.Size()
+		if err != nil {
+			volSize = 0
+		}
+		report.Size = volSize
+		report.Id = vol.Name()
+		var timeout *uint
+		if err := r.RemoveVolume(ctx, vol, false, timeout); err != nil {
+			if !errors.Is(err, define.ErrVolumeBeingUsed) && !errors.Is(err, define.ErrVolumeRemoved) {
+				report.Err = err
+			} else {
+				// We didn't remove the volume for some reason
+				continue
+			}
+		} else {
+			vol.newVolumeEvent(events.Prune)
+		}
+		preports = append(preports, report)
+	}
+	return preports, nil
+}
+
+// SetVolumePinned sets the pinned status of a volume by name.
+// Pinned volumes are excluded from system prune operations by default.
+func (r *Runtime) SetVolumePinned(volumeName string, pinned bool) error {
```

Comment:
IMO, remove this - it doesn't add much over doing `runtime.GetVolume()` yourself and then calling `SetPinned()` on that